Realm/RLMSyncPermissionResults.mm

```diff
@@ -0,0 +1,160 @@
+#import "RLMCollection_Private.hpp"
+#import "RLMObjectSchema_Private.hpp"
+#import "RLMQueryUtil.hpp"
+#import "RLMSchema_Private.hpp"
+#import "RLMSyncPermissionValue_Private.hpp"
+#import "RLMSyncUser_Private.hpp"
+#import "RLMUtil.hpp"
+
+using namespace realm;
+
+@interface RLMSyncPermissionResults () {
+    std::unique_ptr<PermissionResults> _results;
+}
+@end
+
+@implementation RLMSyncPermissionResults
+
+- (NSInteger)count {
+    REALM_ASSERT_DEBUG(_results);
+    return _results->size();
+}
+
+- (RLMNotificationToken *)addNotificationBlock:(RLMPermissionStatusBlock)block {
+    REALM_ASSERT_DEBUG(_results);
+    auto token = _results->async(RLMWrapPermissionStatusCallback(block));
+    return [[RLMCancellationToken alloc] initWithToken:std::move(token) realm:nil];
+}
+
+- (RLMSyncPermissionValue *)objectAtIndex:(NSInteger)index {
+    REALM_ASSERT_DEBUG(_results);
+    try {
+        return [[RLMSyncPermissionValue alloc] initWithPermission:_results->get(index)];
+    } catch (std::exception const& ex) {
+        @throw RLMException(ex);
+    }
+}
+
+- (instancetype)initWithResults:(std::unique_ptr<PermissionResults>)results {
+    if (self = [super init]) {
+        REALM_ASSERT_DEBUG(results);
+        _results = std::move(results);
+    }
+    return self;
+}
+
+- (RLMSyncPermissionValue *)firstObject {
+    return self.count == 0 ? nil : [self objectAtIndex:0];
+}
+
+- (RLMSyncPermissionValue *)lastObject {
+    return self.count == 0 ? nil : [self objectAtIndex:(self.count - 1)];
+}
+
+- (NSInteger)indexOfObject:(RLMSyncPermissionValue *)object {
+    for (int i=0; i<self.count; i++) {
+        if ([[self objectAtIndex:i] isEqual:object]) {
+            return i;
+        }
+    }
+    return NSNotFound;
+}
+
+- (RLMSyncPermissionResults *)objectsWithPredicate:(NSPredicate *)predicate {
+    REALM_ASSERT_DEBUG(_results);
+    auto& results = _results->results();
+    auto query = RLMPredicateToQuery(predicate,
+                                     [RLMObjectSchema objectSchemaForObjectStoreSchema:results.get_object_schema()],
+                                     [RLMSchema dynamicSchemaFromObjectStoreSchema:results.get_realm()->schema()],
+                                     results.get_realm()->read_group());
+    auto filtered_results = std::make_unique<PermissionResults>(_results->filter(std::move(query)));
+    return [[RLMSyncPermissionResults alloc] initWithResults:std::move(filtered_results)];
+}
+
+- (RLMSyncPermissionResults *)sortedResultsUsingProperty:(RLMSyncPermissionResultsSortProperty)property
+                                               ascending:(BOOL)ascending {
+    REALM_ASSERT_DEBUG(_results);
+    auto& results = _results->results();
+    std::string property_name;
+    switch (property) {
+        case RLMSyncPermissionResultsSortPropertyPath:
+            property_name = "path";
+            break;
+        case RLMSyncPermissionResultsSortPropertyUserID:
+            property_name = "user_id";
```

Comment:
are you sure this is `user_id`? Looks like it's actually `userId` according to this: https://github.com/realm/realm-object-store/blob/956bc716aab38719f5a1410a535f14eebf4e7038/src/sync/sync_permission.cpp#L234